src/d3a/d3a_core/simulation.py

```diff
@@ -345,7 +345,8 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
             self.area.cycle_markets()
 
             if self.simulation_config.external_connection_enabled:
-                self.global_objects.update()
+                self.global_objects.update(current_market_slot=self.area.current_market.time_slot
+                                           if self.area.current_market is not None else None)
```

Comment:
Because I have seen this repeating check a lot in our code, how about exposing a small property in the area, namely `current_market_time_slot` that contains this None check and does not leave the responsibility to the caller code ?

Reply:
good idea! Will add it.

Reply:
Actually, I had a look. There are a lot of checks if the current_market is None, but the consequence is different every time. The time_slot is not needed apart from here. Maybe I missed this, in which case, please point me to the places. Thanks!

Reply:
In this case then you can skip this, all good.